exercises/edward_brown/ex01_basics/src/Punctuation.cpp

```diff
@@ -0,0 +1,16 @@
+#include "Punctuation.h"
+#include <algorithm>
+
+bool isPunctuation(char Candidate) {
+  auto const punctuation = std::string(".,?'\\\"!():");
```

Comment:
For *local* variables I tend to prefer the use of `auto` over direct initialization for the following reasons:
1. Less issues with the [most vexing parse](https://en.wikipedia.org/wiki/Most_vexing_parse). Which is especially useful when the LLVM style guide explicitly outlaws uniform initialization for most classes. 
2. It's harder to end up with any uninitialized variables since `auto` variables must have an initializing expression.
3. Where one variable is initialized by a more complex expression or just another variable it will be updated automatically as the type/type name changes.  (If I want to commit to a specific type I can always use a `static_cast<>`).

As an additional nicety when declaring multiple locals on consecutive lines, the variable names line up consistently which I feel improves readability but I accept may just be personal taste...

```
auto firstName = std::string("Joe");
auto surname = std::string("Bloggs");
auto age = 27;

// rather than

std::string firstName("Joe");
std::string surname("Bloggs");
int age = 21;
```
 However, obviously when dealing with non STL library types which may be expensive to move/copy I understand that some extra care must be taken.